lib/Cryptol.cry

```diff
@@ -323,3 +324,87 @@ groupBy : {each,parts,elem} (fin each) =>
   [parts * each] elem -> [parts][each]elem
 groupBy = split`{parts=parts}
 
+/**
+ * Logical implication
+ */
+(==>) : {a} a -> a -> a
+a ==> b = not a || b
+
+/**
+ * Logical negation
+ */
+not : {a} a -> a
+not a = ~ a
+
+/**
+ * Conjunction
+ */
+and : {n} (fin n) => [n] -> Bit
+and xs = ~zero == xs
+
+/**
+ * Disjuntion
+ */
+or : {n} (fin n) => [n] -> Bit
+or xs = zero != xs
+
```

Comment:
Might it make sense to specialize these first four operations to `Bit` values? 

I suppose `and` and `or` already are, but when I'm dealing with a sequence of bits that I want to treat more as "list of bits" than "word", I like to expand the type abbreviation `[n]` -> `[n]Bit`

Also, typo in "Disjuntion"